Uniswap/web3-react
# feat: add support for `optionalChains` in WalletConnect V2

Adds better handling for optional chains as a part of WalletConnect V2. Also, it adds a lot of unit tests to already written code and extends with a series of new ones, to ensure correct handling of default/optional/required chains. It also updated WalletConnect dependency to latest and adds `showQrModal` which is now required.

To test:
- Start localhost example or go to Vercel preview
- Use Uniswap Wallet (dev) build
- Connect
- You should be able to switch across all the chains, because Uniswap Wallet accepts and supports all options chains
- Disconnect
- Connect with Trust Wallet
- You won't be able to switch to anything but main net, because TrustWallet doesn't support "optionalChains" functionality at all yet
